libs/ui/src/invalid_card_screen.tsx

```diff
@@ -1,47 +1,75 @@
 import { DippedSmartCardAuth, InsertedSmartCardAuth } from '@votingworks/types';
 
-import { fontSizeTheme } from './themes';
 import { Main } from './main';
 import { Prose } from './prose';
-import { Screen } from './screen';
 import { RotateCardImage } from './rotate_card_image';
+import { Screen } from './screen';
+import { fontSizeTheme } from './themes';
 import { H1, P } from './typography';
 
-type LoggedOutReason =
-  | DippedSmartCardAuth.LoggedOut['reason']
-  | InsertedSmartCardAuth.LoggedOut['reason'];
+type ReasonAndContext = Pick<
+  DippedSmartCardAuth.LoggedOut | InsertedSmartCardAuth.LoggedOut,
+  'reason' | 'cardJurisdiction' | 'cardUserRole' | 'machineJurisdiction'
+>;
 
 export interface Props {
-  reason: LoggedOutReason;
+  reasonAndContext: ReasonAndContext;
   recommendedAction?: string;
 }
 
 export function InvalidCardScreen({
-  reason,
+  reasonAndContext,
   recommendedAction: recommendedActionOverride,
 }: Props): JSX.Element {
+  const { cardJurisdiction, cardUserRole, machineJurisdiction, reason } =
+    reasonAndContext;
+
   let graphic: JSX.Element | null = null;
   let heading = 'Invalid Card';
   let errorDescription = '';
   let recommendedAction =
-    'Please insert a valid Election Manager or System Administrator card.';
-
-  if (reason === 'card_error') {
-    graphic = <RotateCardImage />;
-    // The only cause we currently know of for a card error
-    heading = 'Card is Backwards';
-    recommendedAction = 'Remove the card, turn it around, and insert it again.';
-  }
-
-  if (reason === 'election_manager_wrong_election') {
-    errorDescription =
-      'The inserted Election Manager card is programmed for another election ' +
-      'and cannot be used to unlock this machine.';
-  }
+    recommendedActionOverride ?? 'Please insert a valid card.';
 
-  if (reason === 'machine_not_configured') {
-    errorDescription =
-      'This machine is unconfigured and cannot be unlocked with this card.';
+  switch (reason) {
+    case 'card_error': {
+      graphic = <RotateCardImage />;
+      // The only cause we currently know of for a card error
+      heading = 'Card is Backwards';
```

Comment:
I think I hit this case when I had a faulty card reader (or faulty connection to the card reader), where the card reader was detected on VxMarkScan but couldn't read any cards. Replacing the card reader solved the problem.

Still, if a backwards card is much more likely to happen than a faulty card reader, makes sense to keep this as is.

Reply:
Ah right, thanks for reminding me of that. I've left the code as is but updated the comment: https://github.com/votingworks/vxsuite/pull/4218/commits/9f3acae111a4c0b8b0cb5caae62d8dba036b4aa6